src/engine/surface_load.c

```diff
@@ -718,11 +716,17 @@ void load_object_collision_model(void) {
     if (o->oCollisionDistance > o->oDrawingDistance) {
         o->oDrawingDistance = o->oCollisionDistance;
     }
+    
+    s32 isLaterallyClose = (sqrLateralDist < sqr(o->oCollisionDistance));
+    s32 isNotBellow = (verticalMarioDiff > 0 || verticalMarioDiff > -o->oCollisionDistance);
+    s32 isNotAbove = (verticalMarioDiff < 0 || verticalMarioDiff < o->oCollisionDistance + 2000.f) ;
+
+    s32 inColRadius = isLaterallyClose && isNotAbove && isNotBellow;
```

Comment:
I would make this into a single statement, so that less useless comparisons need to happen if the first ones fail.

```suggestion
s32 inColRadius = (
       (sqrLateralDist < sqr(o->oCollisionDistance))
    && (verticalMarioDiff > 0 || verticalMarioDiff > -o->oCollisionDistance)
    && (verticalMarioDiff < 0 || verticalMarioDiff < o->oCollisionDistance + 2000.f)
);
```

Reply:
I'm actually not sure about if it would or wouldn't, though now that I think about it, the 'optimal' thing to do would just be to cram it into the if statement at the cost of readability. Might be worth it for this function.

Reply:
Order of likelihood and runtime costs for each check are also important considerations to keep in mind